scripts/signing_secure_boot_dev.sh

```diff
@@ -0,0 +1,152 @@
+EFI_ARCH=x64
+fi
+
+# efi-sign.sh is used to sign: shim, mmx, grub, and kernel (vmlinuz)
+EFI_SIGNING=scripts/efi-sign.sh
+
+######################
+## shim & mmx signing
+######################
+
+# shim dirs
+SHIM_DIR_SRC=$FS_ROOT/usr/lib/shim
+MMX_EFI_SRC=$SHIM_DIR_SRC/mm${EFI_ARCH}.efi
+SHIMX_EFI_SRC=$SHIM_DIR_SRC/shim${EFI_ARCH}.efi
+
+# clean old files
+clean_file ${SHIMX_EFI_SRC}-signed
+clean_file ${MMX_EFI_SRC}-signed
+clean_file $FS_ROOT/boot/shim${EFI_ARCH}.efi
+clean_file $FS_ROOT/boot/mm${EFI_ARCH}.efi
+
+# clean old shim & mmx files in the env
+sudo rm -f ${SHIMX_EFI_SRC}-signed ${MMX_EFI_SRC}-signed \
+           $FS_ROOT/boot/shim${EFI_ARCH}.efi $FS_ROOT/boot/mm${EFI_ARCH}.efi
+
+echo "signing shim${EFI_ARCH}.efi & mm${EFI_ARCH}.efi from location: ${SHIM_DIR_SRC} .."
+sudo ${EFI_SIGNING} $PEM_PRIV_KEY $PEM_CERT ${SHIMX_EFI_SRC} ${SHIMX_EFI_SRC}-signed
+sudo ${EFI_SIGNING} $PEM_PRIV_KEY $PEM_CERT ${MMX_EFI_SRC} ${MMX_EFI_SRC}-signed
+
+# cp shim & mmx signed files to boot directory in the fs.
+sudo cp ${SHIMX_EFI_SRC}-signed $FS_ROOT/boot/shim${EFI_ARCH}.efi
+sudo cp ${MMX_EFI_SRC}-signed $FS_ROOT/boot/mm${EFI_ARCH}.efi
+
+# verifying signature of mm & shim efi files.
+sudo bash scripts/secure_boot_signature_verification.sh -c $PEM_CERT -e $FS_ROOT/boot/shim${EFI_ARCH}.efi  
+sudo bash scripts/secure_boot_signature_verification.sh -c $PEM_CERT -e $FS_ROOT/boot/mm${EFI_ARCH}.efi
+
+######################
+## grub signing
+######################
+
+# clean old files
+clean_file ${GRUB_EFI_SRC}-signed
+clean_file $FS_ROOT/boot/grub${EFI_ARCH}.efi
+
+GRUB_DIR_SRC=$FS_ROOT/usr/lib/grub/x86_64-efi/monolithic/
+GRUB_EFI_SRC=$GRUB_DIR_SRC/grub${EFI_ARCH}.efi
+
+# clean old grub files in the env
+sudo rm -f ${GRUB_EFI_SRC}-signed $FS_ROOT/boot/grub${EFI_ARCH}.efi
+
+echo "signing grub${EFI_ARCH}.efi from location: ${GRUB_EFI_SRC} .."
+sudo ${EFI_SIGNING} $PEM_PRIV_KEY $PEM_CERT ${GRUB_EFI_SRC} ${GRUB_EFI_SRC}-signed
+
+# cp signed grub to fs boot dir.
+sudo cp ${GRUB_EFI_SRC}-signed $FS_ROOT/boot/grub${EFI_ARCH}.efi
+
+# verifying signature of grub efi file.
+sudo bash scripts/secure_boot_signature_verification.sh -c $PEM_CERT -e $FS_ROOT/boot/grub${EFI_ARCH}.efi
+
+######################
+## vmlinuz signing
+######################
+
+CURR_VMLINUZ=$FS_ROOT/boot/vmlinuz-${LINUX_KERNEL_VERSION}-${CONFIGURED_ARCH}
+
+# clean old files
+clean_file ${CURR_VMLINUZ}-signed
+
+# clean old grub files in the env
+sudo rm -f ${CURR_VMLINUZ}-signed
+
+echo "signing ${CURR_VMLINUZ} .."
+sudo ${EFI_SIGNING} $PEM_PRIV_KEY $PEM_CERT ${CURR_VMLINUZ} ${CURR_VMLINUZ}-signed
+
+# rename signed vmlinuz with the name vmlinuz without signed suffix
+sudo cp ${CURR_VMLINUZ}-signed ${CURR_VMLINUZ}
+
+sudo bash scripts/secure_boot_signature_verification.sh -c $PEM_CERT -e ${CURR_VMLINUZ}
+
+#########################
+# Kernel Modules signing
+#########################
+sudo bash scripts/signing_kernel_modules.sh $LINUX_KERNEL_VERSION ${PEM_CERT} ${PEM_PRIV_KEY}
```

Comment:
i do not see module dir provided as required by the script, why?

Reply:
I added the parameter, in general, if you don't, it will sign all the kernel modules from the CWD